auth0/resource_auth0_hook.go

```diff
@@ -178,8 +178,10 @@ func buildHook(d *schema.ResourceData) *management.Hook {
 	return hook
 }
 
-func validateHookNameFunc() schema.SchemaValidateFunc {
-	return validation.StringMatch(
-		regexp.MustCompile("^[^\\s-][\\w -]+[^\\s-]$"),
-		"Can only contain alphanumeric characters, spaces and '-'. Can neither start nor end with '-' or spaces.")
+func validateHookName() schema.SchemaValidateDiagFunc {
+	hookNameValidation := validation.StringMatch(
+		regexp.MustCompile(`^[^\s-][\w -]+[^\s-]$`),
```

Comment:
Raw string literals use backticks (`) instead of quotes (") and thus, do not support any escape sequences. This means that the backslash (\) can be used freely, without the need of escaping.

Since regular expressions have their own escape sequences, raw strings can improve their readability.

Bad pattern:
```go
regexp.Compile("\\A(\\w+) profile: total \\d+\\n\\z")
```

Good pattern:
```go
regexp.Compile(`\A(\w+) profile: total \d+\z`)
```

This was raised by the linter:

S1007: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (gosimple)